actions/image/crop.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Image crop action
+ */
+
+$guid = get_input('guid');
+$entity = get_entity($guid);
+
+if (!($entity instanceof ElggEntity) || !$entity->canEdit()) {
+	register_error(elgg_echo('image:crop:fail'));
+	forward(REFERER);
+}
+
+$x1 = (int) get_input('x1', 0);
+$y1 = (int) get_input('y1', 0);
+$x2 = (int) get_input('x2', 0);
+$y2 = (int) get_input('y2', 0);
+
+$filehandler = new ElggFile();
+$filehandler->owner_guid = $entity->getGUID();
+$filehandler->setFilename("icon/master.jpg");
+$filename = $filehandler->getFilenameOnFilestore();
+
+// Ensure that the image exists in the first place
+if (!file_exists($filename)) {
+	register_error(elgg_echo('image:crop:fail'));
+	forward(REFERER);
+}
+
+$icon_sizes = elgg_get_config('icon_sizes');
+unset($icon_sizes['master']);
+
+// get the images and save their file handlers into an array
+// so we can do clean up if one fails.
+$files = array();
+foreach ($icon_sizes as $name => $size_info) {
+	$resized = get_resized_image_from_existing_file($filename, $size_info['w'], $size_info['h'], $size_info['square'], $x1, $y1, $x2, $y2, $size_info['upscale']);
+
+	if ($resized) {
```

Comment:
It would be great to call some deleteIcons() method on failure and do the cleanup.